lua/tsc/init.lua

```diff
@@ -167,6 +198,10 @@ function M.setup(opts)
   vim.api.nvim_create_user_command("TSC", function()
     M.run()
   end, { desc = "Run `tsc` asynchronously and load the results into a qflist", force = true })
+
+  if config.flags.watch and config.flags.project then
+    M.run()
+  end
```

Comment:
Is there a reason that we need `watch` _and_ `project` to be set? In addition, lets not autostart the plugin in watch mode (especially if we're not in a TypeScript project), lets rather add documentation on how to add an auto command the user to setting up an auto command that invokes our `TSC` command on buffer enter of a `typescript` file. 

We could also potentially make this a configuration option - something like `auto_start_watch_mode` that will noop if the `watch` flag is false, otherwise it'll create the aforementioned auto command on behalf of the user.